woocommerce/google-listings-and-ads
# Guzzle Conflict with other plugins

### Changes proposed in this Pull Request:

This PR ensures that we use a unique namespace for the Guzzle library so we do not conflict with any other plugins that might be using different versions of the same library. Since we not only use the Guzzle library directly, but it's also used by the Google Content API and the Google Ads library, this PR adds some code for replacing namespaces in dependent libraries.

The following points are done in this PR:
- Use a unique namespace for Google\Auth
- Use a unique namespace for GuzzleHttp
- Add dependencies for each of the above two
- Prefix the namesapce for direct usage in the code (ex. `\GuzzleHttp\Psr7\stream_for`)
- When prefixing `namespace` and `use` statements, match the whole word (prevents Google\Auth from matching Google\AuthHandler)
- Removes incompatibility warning for Bookings

Closes #751 


### Detailed test instructions:

1. Checkout PR
2. run `rm -rf vendor && composer install`
3. Use the connection test page to check requests sent directly to WCS
4. Use the connection test page to send proxied requests to the Content API
5. Use the connection test page to send proxied requests to the Ads API library
6. Activate the Bookings plugin and confirm we no longer see an admin warning

Ideally those three types of requests should function without any errors or warnings. However the changes are quite substantial, so ideally we would test any requests that gets sent to an API (through Guzzle).

I've tested with the following list of conflicting plugins, which would previously give an error upon activation:
https://woocommerce.com/products/woocommerce-bookings/
https://wordpress.org/plugins/backwpup/
https://wordpress.org/plugins/simple-sales-tax/
https://wordpress.org/plugins/mailjet-for-wordpress/


### Changelog entry
* Fix - Resolve Guzzle Conflict with other plugins.
